tests/unit/test_instance.py

```diff
@@ -31,7 +31,10 @@ async def test_Instance_init() -> None:
     Test to check whether the __init__ method of Instance
     can tell if the instance URI that's passed in is formatted correctly.
     """
-    keys = generate_keys()
+    event_loop = asyncio.get_running_loop()
+    keys = asyncio.wrap_future(
+        asyncio.run_coroutine_threadsafe(generate_keys(), event_loop), loop=event_loop
+    )
```

Comment:
Hi @jackwotherspoon , I have removed the separate Async loops. 

But one doubt, since the keys are being passed to the Instance object as future, due to double await, the Tests are throwing error that "an awaited expression can't be awaited again". So I removed the await for the generate_keys() function so finally it's being awaited once in the main Instance loop. 

Tell me if it's fine otherwise we can revert it back.